### DISCUSSION TITLE
Welcome to olarm-ha-integration Discussions!

### DISCUSSION THREAD

**Discussion by rainepretorius** [rainepretorius/olarm-ha-integration#8] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by MrPrisoner** [comment]
Hi,

I've been waiting for Olarm integration since I got my unit about a year ago, awesome work, thanks!

I have an IDS x64 panel with 2 partitions (inside, outside), and this integration is working flawlessly. Happy to test new versions for you if you need someone with an IDS panel.

I'm mainly using it to arm/disarm or to check arm status. I created a small automation to turn all the lights on when the alarm triggers. I'm planning to build some more automations for auto arming, maybe even switching off lights when no movement detected in 10 minutes, etc.

Cheers :)

**Reply by pickwit** [reply]
Hi,

I have HACS installed on Home Assistant, however I cannot find the Olarm integration when searching for it. Please share How I can manually install it on HA after downloading the file.

**Comment by rainepretorius** [comment]
Copy it to the custom_components folder on your server.

**Comment by pickwit** [comment]
Thank you!

**Comment by MrPrisoner** [comment]
Hi pickwit,

You can add the Olarm Sensors repo to custom repos on HACS > Integrations > tripple dot menu.

Probably easier than manually copying to custom_components, and should allow updates via the UI.

**Reply by rainepretorius** [reply]
That should work a lot better.

**Comment by TGouws** [comment]
Hi Rain, will it be possible to add the panic button to your integration?

**Reply by rainepretorius** [reply]
I will see if I can integrate that feature.

**Reply by rainepretorius** [reply]
It seems that it is not possible as Olarm has no public API to trigger that function. See here for a related query that might be of assistance. https://github.com/rainepretorius/olarm-ha-integration/discussions/14

**Reply by TGouws** [reply]
Thank you Rain will check

**Comment by BeeSteady** [comment]
Hi all! So I’d like to start off with saying WOW - thank you and well done! I am an absolute noob, know next to nothing about programming and have only been playing with HA for about 2 days. I cannot seem to find the integration - how should i go about installing it?
Any help would be greatly appreciated!

**Reply by MrPrisoner** [reply]
Hi BeeSteady, you need to add this repo as a Custom Repository on HACS. Do you have HACS installed on HA? If so you can go to HACS > Integrations > tripple dot menu > Custom Repositories, paste the url of this repo (https://github.com/rainepretorius/olarm-ha-integration) and Category of Integration. Then follow the steps as listed on this repo home page.

**Comment by BeeSteady** [comment]
Thanks so much! You’re a lifesaver. Now just waiting on Olarm’s poor service in getting me API accessJ WaltersBeeSteady Productionswww.beesteadyproductions.comOn 28 Apr 2023, at 10:07 AM, MrPrisoner ***@***.***> wrote:﻿
Hi BeeSteady, you need to add this repo as a Custom Repository on HACS. Do you have HACS installed on HA? If so you can go to HACS > Integrations > tripple dot menu > Custom Repositories, paste the url of this repo (https://github.com/rainepretorius/olarm-ha-integration) and Category of Integration. Then follow the steps as listed on this repo home page.

**Comment by BeeSteady** [comment]
Thanks so much!!!Little question here - we have an installer account with multiple sites. How do you specify which one? And what should be filled in here (see picture)Thanks!J WaltersBeeSteady Productionswww.beesteadyproductions.comOn 28 Apr 2023, at 10:07 AM, MrPrisoner ***@***.***> wrote:﻿
Hi BeeSteady, you need to add this repo as a Custom Repository on HACS. Do you have HACS installed on HA? If so you can go to HACS > Integrations > tripple dot menu > Custom Repositories, paste the url of this repo (https://github.com/rainepretorius/olarm-ha-integration) and Category of Integration. Then follow the steps as listed on this repo home page.

**Reply by rainepretorius** [reply]
Hi, I cannot see the pictures.

**Comment by BeeSteady** [comment]
Apologies - it refers to the fields under the API access on Olarm’s site. The webhook secret and webhook field.In home assitant i get the error “failed to set up”J WaltersBeeSteady Productionswww.beesteadyproductions.comOn 28 Apr 2023, at 12:03 PM, Raine Pretorius ***@***.***> wrote:﻿
Hi, I cannot see the pictures.

**Reply by rainepretorius** [reply]
The intecration does not use Olarms Webhook or the secret. So it should not cause issues.

**Comment by BeeSteady** [comment]
Thanks! Considering the fact that there are multiple sites on the account, how do i specify a specific one? Issue is even without that, it doesn’t load in HA. I can share the logs?J WaltersBeeSteady Productionswww.beesteadyproductions.comOn 28 Apr 2023, at 12:20 PM, Raine Pretorius ***@***.***> wrote:﻿
The intecration does not use Olarms Webhook or the secret. So it should not cause issues.

**Reply by rainepretorius** [reply]
The logs would assist a lot with debugging. Please make sure you enable log debugging on the Devices and Integrations page. The integration currently does not allow you to specify a device/site and loads all of them.

**Comment by rainepretorius** [comment]
Please note that I will have semester tests from Wednesday(03 May 2023 till the 29th of May 2023. I will not reply to any new issues and the development for fixes of existing problems will be halted until my tests are over.

**Reply by JTUBB69** [reply]
Good luck!

**Comment by neilhowieson** [comment]
I am looking at getting an Olarm and then obviously integrating to HA with this integration. I have just been reading a lot of negative reviews on the iOS version of the app with people saying they often aren't able to connect via the app. Has anyone here battled with that when using HA? Just trying to figure out if it's an app issue or a general Olarm issue...

**Reply by MrPrisoner** [reply]
I've had my Olarm unit now for about a year and in that time I've very rarely had issues. Olarm's comms about issues are also quite good, you get emails about the status of the incident and usually resolved in a couple of hours. And most incidents were related to mobile network problems, if you have WiFi also connected then it shouldn't affect you.

I can't remember the last time I used my alarm's keypad, I do everything via the Olarm Android app or HA. The HA integration is quite stable now thanks to @rainepretorius, there is just a few seconds delay sometimes in refreshing the state of entities due to Olarm's API limiter. They are in discussions about webhooks/realtime updates. But the delay is barely noticeable, I have some automations to switch on garden lights when beams pick up movement and it works like a charm.

So in a nutshell, I would recommend Olarm based on my experience.

**Reply by rwoeke** [reply]
I can second Olarm is very stable platform and worth every sent. Its an absolute no Brainer!
HA integration also works well now!

**Reply by neilhowieson** [reply]
Thanks for your advice. I have taken the plunge and installed the system. Now I am just waiting for the API access. I logged a ticket as suggested, but the response is really slow. Do any of you have a contact to try to speed this up?

**Reply by rwoeke** [reply]
Hi, You can just log into olarm app and get an api key from there. top right menu - 
API access
